processing/src/main/java/org/apache/druid/query/CacheStrategy.java

```diff
@@ -48,21 +50,27 @@
   boolean isCacheable(QueryType query, boolean willMergeRunners);
 
   /**
-   * Computes the cache key for the given query
+   * Computes the per-segment cache key for the given query. Because this is a per-segment cache key, it should only
+   * include parts of the query that affect the results for a specific segment (i.e., the results returned from
+   * {@link QueryRunnerFactory#createRunner}).
    *
-   * @param query the query to compute a cache key for
+   * @param query the query to be cached
    *
-   * @return the cache key
+   * @return the per-segment cache key
    */
   byte[] computeCacheKey(QueryType query);
 
   /**
-   * Computes the result level cache key for the given query.
-   * Some implementations may include query parameters that might not be used in {@code computeCacheKey} for same query
+   * Computes the result-level cache key for the given query. The result-level cache will tack on datasource and
+   * interval details, so this key does not need to incluce datasource and interval. But it should include anything
```

Comment:
```suggestion
   * interval details, so this key does not need to include datasource and interval. But it should include anything
```

^ trying out github suggestions, supposedly it lets you accept a change without needing to make the change locally :)

 https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes

Reply:
Whoa, cool. I clicked the button and it seemed to work.